src/value.jl

```diff
@@ -87,9 +87,9 @@ Base.convert(::Type{Union{T, Nothing}}, x::T) where {T <: CatValue} = x
 Base.convert(::Type{S}, x::T) where {S, T <: CatValue} =
     T <: S ? x : convert(S, get(x))
 Base.convert(::Type{Union{S, Missing}}, x::T) where {S, T <: CatValue} =
-    T <: S ? x : convert(S, get(x))
+    T <: Union{S, Missing} ? x : convert(Union{S, Missing}, get(x))
```

Comment:
Are these two definitions for `Nothing` and `Missing` needed? It seems that by leaving https://github.com/JuliaData/CategoricalArrays.jl/pull/213/files#diff-0c97238b80ec111debf6823967ed4c9eR87 only (ie. the default `convert`) we should get exactly the same?

Reply:
Yes, that's due to ambiguities with methods defined in Base.

Reply:
OK, this is the pain I have with DataFrames.jl every time I do a major redesign of something that touches Base.

Reply:
Hopefully with https://github.com/JuliaData/CategoricalArrays.jl/pull/201 we will be able to simplify this.